plugins/modules/aci_l2out.py

```diff
@@ -0,0 +1,309 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020, Sudhakar Shet Kudtarkar (@kudtarkar1)
+# Copyright: (c) 2020, Shreyas Srish <ssrish@cisco.com>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: aci_l2out
+short_description: Manage layer2(L2Out) objects.
+description:
+- Manage outer layer2 on Cisco ACI fabric.
+options:
+  tenant:
+    description:
+    - Name of existing tenant.
+    type: str
+  l2out:
+    description:
+    - The name of outer layer2.
+    type: str
+    aliases: [ 'name' ]
+  description:
+    description:
+    - The description of outer layer2.
+    type: str
+  bd:
+    description:
+    - Name of the Bridge domain which is associted with L2Out.
+    type: str
+  domain:
+    description:
+    - Name of the external L2 Domain that is being associated with L2Out.
+    type: str
+  vlan:
+    description:
+    - Vlan which being associated with L2
+    type: str
+  targetDscp:
+    description:
+    - target what
+    type: str
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    type: str
+    choices: [ absent, present, query ]
+    default: present
+extends_documentation_fragment:
+- cisco.aci.aci
+
+notes:
+- The C(tenant), C(bd) must exist before using this module in your playbook.
+  The M(aci_tenant), M(aci_bd) modules can be used for this.
+seealso:
+- name: APIC Management Information Model reference
+  description: More information about the internal APIC class B(tenant), B(bd).
+  link: https://developer.cisco.com/docs/apic-mim-ref/
+author:
+- kudtarkar1 (@kudtarkar1)
+'''
+
+EXAMPLES = r'''
+- name: Add New L2Out
+  cisco.aci.aci_l2out:
+    host: apic
+    username: admin
+    password: SomeSecretePassword
+    tenant: Auto-Demo
+    l2out: l2out
+    description: via Ansible
+    bd: bd1
+    domain: l2Dom
+    vlan: vlan-3200
+    state: present
+    delegate_to: localhost
+
+- name: Remove L2Out
+  cisco.aci.aci_l2out:
+    host: apic
+    username: admin
+    password: SomeSecretePassword
+    tenant: Auto-Demo
+    l2out: l2out
+    bd: bd1
+    domain: l2Dom
+    vlan: vlan-3200
+    state: absent
+    delegate_to: localhost
+
+- name: Query L2Out
+  cisco.aci.aci_l2out:
+    host: apic
+    username: admin
+    password: SomeSecretePassword
+    l2out: l2out
+    state: query
+    delegate_to: localhost
+    register: query_result
+
+- name: Query all
```

Comment:
This does not looks like it would work when I read the code. Does this work? What is the return value?
If you want to return all L2Outs, we should allow to filter by tenants or get all of them.

Reply:
Also the name of this task is incomplete.